code/mob/living/critter.dm

```diff
@@ -343,6 +347,23 @@ ABSTRACT_TYPE(/mob/living/critter)
 		src.ghostize()
 		qdel (src)
 
+	proc/remove_arm(var/mob/living/critter/M, var/arm)
+		if(arm == "left")
+			var/datum/handHolder/HH = hands[1]
+			qdel(HH)
+			new M.left_arm(M.loc)
+			M.update_dead_icon()
+			return
+		if(arm == "right")
+			var/datum/handHolder/HH = hands[2]
```

Comment:
I'm not sure this works. A lot of critters have a "mouth" limb in slot 2 on handholders.

Reply:
It works for brullbars and should work for bears which is what its intended for currently

Reply:
Maybe throw a null check on there for the arms? Better safe than runtime

Reply:
I wouldn't really call this resolved. I meant the null check should be on `src.left_arm` and `src.right_arm`, since for most critters calling this would throw a runtime. Also the call signature is weird, why are you passing this mob as an argument? I'd basically rewrite this.

```
proc/remove_arm(var/arm_side = "left")
	switch(arm_side)
		if "left"
			if(!src.left_arm)
				return
			var/datum/handHolder/HH = hands[1]
			if(!HH)
				return //in case two action bars are running at the same time, don't duplicate arms
			qdel(HH)
			new src.left_arm(src.loc)
			src.update_dead_icon()
			return
...
```

Reply:
Yeah I had the wrong idea there I will fix the proc